### PR TITLE
feat(hogvm): hogvm bytecode action matching in the plugin server

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Splitting work from https://github.com/PostHog/posthog/pull/16304 - this is the final part.

## Changes

Actually match the HogVM bytecode in the plugin server.

## How did you test this code?

Tests

### CODE REVIEW

**Comment by mariusandra** [comment]
@Twixes This is ready for another look. Currently we're just discarding the HogVM matches, even if they match while the legacy version doesn't. This means I had to update the tests to explicitly call `matchBytecode` to verify that it works, as the normal/legacy `match` discarded the results.

As it is now, this PR won't enable HogQL expression matching. Should we change things and enable/append the HogVM matches to the legacy matches?

File: plugin-server/src/worker/ingestion/action-matcher.ts [hunk 1]
```diff
@@ -137,43 +139,125 @@ export class ActionMatcher {
...
+        const legacyResponse = await this.matchLegacy(event, elements)
+
+        if (
+            legacyResponse.length !== hogVMResponse.length ||
+            !legacyResponse.every((action, index) => action.id == hogVMResponse[index]?.id)
+        ) {
+            if (hogVMResponse.length > legacyResponse.length) {
+                this.statsd?.increment('action_matches_hogvm_legacy_mismatch_hogvm_matched_more')
+            } else {
+                this.statsd?.increment('action_matches_hogvm_legacy_mismatch')
```
Comment (Twixes) [inline plugin-server/src/worker/ingestion/action-matcher.ts:154]:
```suggestion                   
                this.statsd?.increment('action_matches_hogvm_legacy_mismatch_hogvm_matched_less')
```
Very wordy but this is in line with the other metric's naming

**Review by Twixes — APPROVED** [review]
Looks good, let's just use consistent naming for the (mis)match metrics before merging

**Comment by Twixes** [comment]
Let's do a dry run for a day first, if there are no mismatches ideally we'll be able to fully swap to using the bytecode results instead of trying to reconcile the results

**Pull request merged**